### PR TITLE
Patch release guide

### PR DESCRIPTION
**Pull request by JuanPedroGHM**
## Due Diligence

- General:
    - [x]  **base branch** must be `main` for new features, latest release branch (e.g. `release/1.3.x`) for bug fixes
    - [x]  **title** of the PR is suitable to appear in the [Release Notes](https://github.com/helmholtz-analytics/heat/releases/latest)
- Implementation:
    - [x] unit tests: all split configurations tested
    - [x] unit tests: multiple dtypes tested
    - [x] documentation updated where needed

## Description

Updates RELEASE.md with step-by-step guide to releasing a patch version of heat to PyPI and to conda-forge. Steps for releasing mayor and minor versions to conda-forge were added as well. 

Issue/s resolved: #

## Changes proposed:

- Documentation changes in RELEASE.md

## Type of change
- Documentation update

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -54,21 +54,36 @@ extension: str = "dev" # <-- set to None
...
+    - Make sure the SHA points to the correct PyPI release.
+    - Make sure dependencies match.
+  - Once the PR is done, wait for the CI checks to finish and merge.
+  - Refer to the conda-forge docs if there are any issues: https://conda-forge.org/docs/maintainer/updating_pkgs.html#pushing-to-regro-cf-autotick-bot-branch
+
+9. Go back to the Release Notes draft and publish them. The new release is out!
+
+  - Make sure the CHANGELOG.md got updated, if not, call @JuanPedroGHM.
+
+10. Now we want to update `main` to the latest version, and we want the version on `main` to be `1.3.0-dev`.
```
Comment (ClaudiaComito) [inline RELEASE.md:86]:
```suggestion
10. Now we want to update `main` to the latest version, and we want the version on `main` to be `1.4.0-dev`.
```

File: RELEASE.md [hunk 2]
```diff
@@ -78,9 +93,97 @@ extension: str = "dev" # <-- set to None
...
+
+
+13. Get approval and merge. You're done!
 
 ### Patch release
 
-(e.g. 1.3.1 --> 1.3.2)
+(e.g. 1.3.0 --> 1.3.1)
+
+1. Check that all intended branches have been merged to `main`. Branch off from main and create a new release branch:
```
Comment (ClaudiaComito) [inline RELEASE.md:106]:
```suggestion
1. Check that all intended branches have been merged to the release branch you want to upgrade, in this example `release/1.3.x`. Branch off from `release/1.3.x` and create a new branch:
```

File: RELEASE.md [hunk 3]
```diff
@@ -78,9 +93,97 @@ extension: str = "dev" # <-- set to None
...
+  - Squash and merge.
+
+
+5. Draft release notes:
+
+  - Go to the GitHub repo's [Releases](https://github.com/helmholtz-analytics/heat/releases) page.
+  - The release notes draft is automated. Click on Edit Draft
+  - Select the new tag: `v1.3.1`
+  - Edit release notes as needed (see older releases)
+
```
Comment (ClaudiaComito) [inline RELEASE.md:144]: Save but do not publish?

File: RELEASE.md [hunk 4]
```diff
@@ -78,9 +93,97 @@ extension: str = "dev" # <-- set to None
...
+  - Once the PR is done, wait for the CI checks to finish and merge.
+  - Refer to the conda-forge docs if there are any issues: https://conda-forge.org/docs/maintainer/updating_pkgs.html#pushing-to-regro-cf-autotick-bot-branch
+
+
+10. Go back to the Release Notes draft and publish them. The new release is out!
+  - Make sure the CHANGELOG.md got updated in the release branch, in this case `release/1.3.x`, if not, call @JuanPedroGHM.
+
+11. Now we want to update `main` to the latest version, and we want the version on `main` to be `1.4.0-dev`.
+  - Create a new branch from `release/1.3.x`, for example `workflow/merge-latest-release-into-main`.
+  - Create a PR from `workflow/merge-latest-release-into-main`
```
Comment (ClaudiaComito) [inline RELEASE.md:186]:
```suggestion
  - Create a new branch from `release/1.3.x`, for example `merge-latest-release-into-main`
  - Merge `main` into `merge-latest-release-into-main`, resolve conflicts and push.
  - Create a PR from `merge-latest-release-into-main`, base branch must be `main`
```

File: RELEASE.md [hunk 5]
```diff
@@ -78,9 +93,97 @@ extension: str = "dev" # <-- set to None
...
+  - Make sure the version number is corrent in main.
```
Comment (ClaudiaComito) [inline RELEASE.md:187]:
```suggestion
  - Make sure the version number in `merge-latest-release-into-main` is correct (i.e., it matches that in `main`).
```

File: RELEASE.md [hunk 6]
```diff
@@ -78,9 +93,97 @@ extension: str = "dev" # <-- set to None
...
+  - Make sure the CHANGELOG.md is the same in both branches.
```
Comment (ClaudiaComito) [inline RELEASE.md:188]:
```suggestion
  - Make sure the CHANGELOG.md in `merge-latest-release-into-main` matches that in the latest release branch, in this example`release/1.3.x`.
```

**Review by ClaudiaComito — CHANGES_REQUESTED** [review]
Thanks a lot @JuanPedroGHM for all the help last week and for updating the instructions. I've suggested a few changes in the review.

File: RELEASE.md (hunk 3 above)
Comment (JuanPedroGHM) [reply RELEASE.md:145]: :+1:

**Review by ClaudiaComito — DISMISSED** [review]
Great, thanks a lot @JuanPedroGHM !

**Review by ClaudiaComito — APPROVED** [review]
Thanks a lot @JuanPedroGHM !

**Review by ClaudiaComito — APPROVED** [review]

**Pull request merged**